src/celeritas/track/detail/TrackInitAlgorithms.cu

```diff
@@ -93,16 +209,99 @@ void partition_initializers(
 {
     ScopedProfiling profile_this{"partition-initializers"};
 
-    // Partition the indices based on the track initializer charge
-    auto start = device_pointer_cast(init.indices.data());
-    auto end = start + count;
+    // Understandably, celeritas doesn't like creating zero-byte vectors. Since
+    // cub needs some vectors, trying to allocate these leads to a failed
+    // assertion. So, just return. No need to partition zero tracks.
+    if (count == 0)
+        return;
```

Comment:
To ensure consistency between host/device, it might be better to move the check to the caller's responsibility, and put a `CELER_EXPECT(count !=0)` conditional on the function.